keras/src/trainers/data_adapters/generator_data_adapter.py

```diff
@@ -49,7 +49,7 @@ def convert_to_tf(x, spec):
             return x
 
         def get_tf_iterator():
-            for batch in self.generator:
+            for batch in self.generator():
```

Comment:
Why is this necessary -- it's `generator` iterable?

Reply:
This is for fixing https://github.com/keras-team/keras/issues/20402
We need to embed `generator` into a lambda to make it able to reset:
https://github.com/keras-team/keras/blob/fe397ed1af44670a838135a9804fab616f5ed4b8/keras/src/trainers/data_adapters/generator_data_adapter.py#L87